corehq/apps/userreports/adapter.py

```diff
@@ -77,6 +77,13 @@ def save(self, doc, eval_context=None):
         indicator_rows = self.get_all_values(doc, eval_context)
         self._save_rows(indicator_rows, doc)
 
+    def bulk_save(self, docs):
+        """
+        Evalutes UCR rows for given docs and saves the result in bulk.
+        Only supported in SQL
+        """
+        raise NotImplementedError
```

Comment:
I think this could provide a default naive solution that just loops docs to put into save.

Reply:
I guess I have to do it, since the adapter may point to ES.